src/hooks/useCodapState.tsx

```diff
@@ -158,15 +158,22 @@ export const useCodapState = () => {
     position: number, newAttrsOrder: Array<any>) => {
     // first update the local collection state to avoid asynchronous re-render on plugin side
     const collIndex = collections.indexOf(coll);
-    const newCollection = {...collections[collIndex]};
-    newCollection.attrs = newAttrsOrder;
+    const updatedCollection = {...collections[collIndex]};
+    updatedCollection.attrs = newAttrsOrder;
     const newCollections = [...collections];
-    newCollections[collIndex] = newCollection;
+    newCollections[collIndex] = updatedCollection;
     setCollections(newCollections);
 
     await connect.updateAttributePosition(selectedDataSet.name, coll.name, attrName, position);
   };
 
+  const handleAddCollection = async () => {
+    const newCollectionName = `Collection ${collections.length + 1}`;
```

Comment:
I wonder if this will error out if you create two collections via this method and then delete the second to last one created and then try to create again - the collection name will then be a duplicate of the current last collection.

Reply:
I just tested it in CODAP itself and it appends a number if the collection already exists - I'd check if that is done in the API or in the UI code.